src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -221,22 +241,127 @@ object ScalafixPlugin extends AutoPlugin {
   }
 
   private def runArgs(
-      paths: Seq[Path],
-      mainArgs: ScalafixArguments,
-      logger: Logger
+      interface: ScalafixInterface,
+      streams: TaskStreams
   ): Unit = {
-    val finalArgs = mainArgs
-      .withPaths(paths.asJava)
-
+    val paths = interface.args.collect { case Arg.Paths(paths) => paths }.flatten
     if (paths.nonEmpty) {
-      if (paths.lengthCompare(1) > 0) {
-        logger.info(s"Running scalafix on ${paths.size} Scala sources")
+      val cacheKeyArgs = interface.args.collect {
+        case cacheKey: Arg.CacheKey => cacheKey
       }
 
-      val errors = finalArgs.run()
-      if (errors.nonEmpty) {
-        throw new ScalafixFailed(errors.toList)
+      // used to signal that one of the argument cannot be reliably stamped
+      object StampingImpossible extends RuntimeException with NoStackTrace
+
+      implicit val stamper = new CacheKeysStamper {
+        override protected def stamp: Arg.CacheKey => Unit = {
+          case Arg.ToolClasspath(toolClassLoader) =>
+            @tailrec def withParentRec(classLoader: URLClassLoader): Unit = {
+              val urls = classLoader.getURLs
+              // to keep it simple, don't support caching when dealing with directories
+              if (urls.exists(_.toString.endsWith("/")))
+                throw StampingImpossible
```

Comment:
once/if https://github.com/scalacenter/sbt-scalafix/pull/100 gets merged, I will implement this, but for now, it's impossible to reach/test as the `toolClasspath` set programmatically by the plugin (i.e. not via a CLI `--toolclasspath`, handled below in `ParsedArgs` as we don't have an explicit parser for it) only contains JARs.

Reply:
handled in 8b7cb8f